v2 release

myrrc opened the pull request.

PowerUser64 commented:
Is the v2 branch in a state that's ready for testing? I'd be interested in helping test it. (I'm just getting started with reaper-keys.)

If it's ready, how should I install it? I'm noticing there's no index.xml to import to reapack.

myrrc commented:
Unfortunately, it isn't yet. I've had to take a break from development due to some issues, and will get back in a week or two.
Target goal for this package is to be installed via Reapack, though I'd add index.xml once everything is ready (I don't want to pollute commit history with index.xml edits).

Personally, I cloned this branch to `~/.config/REAPER/Scripts/reaper-keys-dev` (Linux) and launched `Scripts/install-reaper-keys.lua` from Reaper.

If you want, I can send you a message once everything is wrapped up

PowerUser64 commented:
Understandable. If you would like me to test, feel free to send a message and/or ping me here! Just know I won't be able to test much more than "does it work?" because I haven't used reaper-keys before (though, I currently use vim and I would like to start using this). I'll wait until this update is finalized so I can build my config around the latest version. (mostly so I'm not porting/swapping to a new version right after getting it.)

lmk when you'd like me to test! :+1:

tkna91 commented:
@myrrc What is your situation? I look forward to your implementation.

myrrc commented:
> @myrrc What is your situation? I look forward to your implementation.

Hey! Fixing bugs locally, not committing as progress is moving very slowly. 

Let's go this way -- by end of this week I'll update this PR with Reapack index and all applied updates so that you and other people could test it.

myrrc commented:
Looks like this KeyMap thing doesn't work at all in Reaper 7. I've updated it now and will look at it thoroughly near the weekend.

myrrc commented:
Autofocus may be solved by https://forum.cockos.com/showthread.php?t=273404

myrrc commented:
@PowerUser64 @tkna91 Ok, so I've fixed the issue (space in script name).
I've updated the PR header with path to dev `index.xml`.

tkna91 commented:
I installed and Run and imported key bindings.
I am using sway, but I think there are a lot of things that don't seem to work right.
Maybe it's the tiling window manager or Wayland.
I'm not sure, so it could be part of the spec.

tkna91 commented:
`space esc space` causes the action list to be displayed.

https://github.com/gwatcha/reaper-keys/assets/102382754/1c9c2896-2b74-4d61-bdaf-4e7e0eedd374

tkna91 commented:
`space g` does not work

https://github.com/gwatcha/reaper-keys/assets/102382754/0f50f9b3-0eb1-4d2c-83fb-bfe61d6db3d6

PowerUser64 commented:
Unfortunately I don't really have time to test this right now :/
I hope the launch goes well! I'll test things out when I get the chance to.

tkna91 commented:
The menu closes after a while after `t`, etc., but the keybindings continue, which is confusing.

https://github.com/gwatcha/reaper-keys/assets/102382754/07ee5f48-6c51-4474-b807-d364c4e7c9d5

myrrc commented:
Thanks for the testing @tkna91, I'll add these bugs to PR description and try to fix them soon.

Don't hesitate to contact me. I'm also using sway on Wayland so our configurations will surely be alike

tkna91 commented:
Personally, I prefer to take my time reading menus, so I would be happier if they did not close.
Or if you could make it adjustable, that would be great.

myrrc commented:
Not sure this is exactly what you want, but there is a parameter https://github.com/gwatcha/reaper-keys/blob/3d6c87dbb9a178bab991bf4c196d30778cf64b54/definitions/gui_config.lua#L25 here which you can set to 0

myrrc commented:
Regarding window size, this is a common issue with the used gui framework. I'll try to rewrite it to gfx, currently you can tweak this at sway side using `for_window [title="Reaper Keys Feedback"]`

myrrc commented:
> * The menu closes after a while after `t`, etc., but the keybindings continue, which is confusing.
> 
> 
> 20231201-143914_screencast.mp4

This isn't a bug, actually. The feedback window is a helper thing if you don't remember specific combinations. You can tweak `hide_after` setting to change it

myrrc commented:
> space esc space causes the action list to be displayed.

Fixed
> space g does not work

Fixed
> The menu closes after a while after t, etc., but the keybindings continue, which is confusing.

I removed auto-hide as it caused a lot of pain to implement properly. There is now an option to dock the window in Docker which is assumed to provided better flexibility

tkna91 commented:
Sorry if this is a problem with my environment.

The cursor position in the input window is wrong.

Screencast:
https://github.com/gwatcha/reaper-keys/assets/102382754/fdba4265-a0b9-418b-86ef-7d97d176e794

tkna91 commented:
PAGEDOWN from input window does not work
pagedown from input window does not work

https://github.com/gwatcha/reaper-keys/assets/102382754/0edaee9e-961f-45d4-9a09-197f49e78635

tkna91 commented:
No scrolling when moving cursor outside of display screen

https://github.com/gwatcha/reaper-keys/assets/102382754/7fe260d0-70a4-4e61-a1ca-4520b365416a

myrrc commented:
> The cursor position in the input window is wrong.

This is an issue with scythe (GUI library). I'll try to look into it

> pagedown from input window does not work

It works now but I didn't propagate changes to index.xml yet, will do tomorrow. However, PdDown is not currenly mapped to any key.

> No scrolling when moving cursor outside of display screen

This is a heisenbug also in scythe. I managed to reproduce it but didn't track it down.

Thanks for the reports!

myrrc commented:
> The cursor position in the input window is wrong.

Generally, issue is that in non-monospaced fonts we can't estimate whole position by extrapolating char width. I've made a fix that for caret we use string width as it is, but selection is pretty much screwed.

myrrc commented:
Noticed a regression for ( -- parses as NumDown

0atman commented:
LOVE that we can dock the feedback window now - thank you! :-D

myrrc commented:
You're welcome :)
